nicolegooden/fitlit-starter-kit
# Hydration implementation ng

#### What's this PR do?

+ Adds appropriate methods to `HydrationRepository.js`
+ Restructures original tests in `Hydration-test.js` to match our new understanding of how the implementation file should be built
+ Applies some syntax changes based on linter suggestions/errors (mostly semicolons)
+ Adds `moment` as an npm package, to be used in `HydrationRepository.js`
+ Resolves merge conflict (occurred in `index.html` and kept KJ's changes from main to override my older version of code)
+ Handles some refactoring of event listeners in `scripts.js`

#### Where should the reviewer start?

Start in `HydrationRepository.js` and `Hydration-test.js` and get a sense of what each test is verifying and how each method is returning what it needs to.  We have some seriously good SRP with our helper functions! I have a feeling we will need to have a similar setup for Iteration 3 ..

#### How should this be manually tested?

Run `npm test` to ensure all tests are passing.  There is no DOM added to `scripts.js` to display any of these methods on `HydrationRepository`, so opening in the browser should not show any updates from our last merge.

#### Any background context you want to provide?

N/A

#### What are the relevant tickets?

Closes #10 
Closes #11 
Closes #13 
Closes #19 

#### Screenshots (if appropriate)

N/A

#### Questions:

Let's think about how we will need to edit our HTML so that there is an input for the date to be selected, then use that day to be passed into our `HydrationRepository` methods where the date parameter exists.  I think this will be the easiest way, what do you think?

Also, this is a reminder to us to run the linter more often!
